packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -347,12 +364,24 @@ module.exports = {
 
     await validateRegisterBody(params);
 
-    const role = await strapi
-      .query('plugin::users-permissions.role')
-      .findOne({ where: { type: settings.default_role } });
+     // Determine role to assign
+    let role;
+    if (ctx.request.body.role) {
```

Comment:
Role should be sent during registration that would be a huge security vulnerability